ios/TextRecognition.swift

```diff
@@ -111,12 +113,23 @@ class TextRecognition: NSObject {
 
       let languageCode = language?.rawValue
 
-      return ["text": recognizedText, "languageCode": languageCode ?? "[Unknown]"]
+      return ["text": recognizedText as String,
+              "languageCode": languageCode ?? "unknown" as String,
+              "confidence": topCandidate.confidence as Any,
+              // "x": observation.boundingBox.origin.x as Any,
+              // "y": observation.boundingBox.origin.y as Any,
+              "leftX": observation.boundingBox.minX as Any,
+              "middleX": observation.boundingBox.midX as Any,
```

Comment:
@ElicaInc I absolutely _love_ these additions. Thank you for this. One question I wanted to run by you — are these technically being returned as `Double`s or `Float`s (or neither?) ?

<img width="310" alt="Screenshot 2023-04-14 at 5 30 13 PM" src="https://user-images.githubusercontent.com/6411509/232160183-36662176-6363-4cc7-81b3-10e8e5dbae01.png">

Reply:
ChatGPT tells us: 
```
In Swift, the as keyword is used for type casting, which allows you to convert an instance of one type to another type. However, not all types can be cast to each other.

In the case of observation.boundingBox.minX, it is a CGFloat type. CGFloat is a floating-point type that is used to represent values that may vary in precision on different platforms.

When you use observation.boundingBox.minX as Any, you are casting the value to the Any type, which is a special type that can represent any value at all. This works because Any is a supertype of all other types in Swift.

When you use Float(observation.boundingBox.minX), you are explicitly converting the CGFloat value to a Float value. This works because Float is a floating-point type that is compatible with CGFloat.

However, when you use observation.boundingBox.minX as Float, you are attempting to cast the value to a Float type directly. This does not work because Float and CGFloat are two distinct types in Swift and are not directly convertible to each other.
```

Reply:
I am not sure whether `observation.boundingBox.minX as Any` is better.

Reply:
I'm not entirely convinced that this is the best option.
```
            return ["text": recognizedText as String,
                    "languageCode": languageCode ?? "unknown" as String,
                    "confidence": topCandidate.confidence as Float, // Any to Float
                    //"x": observation.boundingBox.origin.x as Any,
                    //"y": observation.boundingBox.origin.y as Any,
                     "leftX": Float(observation.boundingBox.minX),
                     "middleX": Float(observation.boundingBox.midX),
                     "rightX": Float(observation.boundingBox.maxX),
                     "bottomY": Float(1 - observation.boundingBox.minY),
                     "middleY": Float(1 - observation.boundingBox.midY),
                     "topY": Float(1 - observation.boundingBox.maxY),
                     "width": Float(observation.boundingBox.width),
                     "height": Float(observation.boundingBox.height),
            ]
```

Reply:
```
export type ResultBlock = {
  ······
  confidence: number // add this line
  ······
}
```

Reply:
Apologies on the delay!

> If you want to explicitly convert a CGFloat value to a Float value, you should use Float(observation.boundingBox.minX). This is a type conversion that is safe and efficient because CGFloat and Float are interchangeable floating-point types in Swift.

Nice! TIL. Swift has some nice conventions. Just to be on the safe-side I confirmed this does appear to be the [preferred means of converting CGFloat to Float over at StackOverflow](https://stackoverflow.com/questions/24147568/casting-cgfloat-to-float-in-swift#answer-24147647)

So I'm good with it! I'll test out the proposed changes tonight and push them up here once I verify 👍